pdns/libssl.hh

```diff
@@ -91,8 +91,17 @@ public:
   ~OpenSSLTLSTicketKey();
 
   bool nameMatches(const unsigned char name[TLS_TICKETS_KEY_NAME_SIZE]) const;
-  int encrypt(unsigned char keyName[TLS_TICKETS_KEY_NAME_SIZE], unsigned char *iv, EVP_CIPHER_CTX *ectx, HMAC_CTX *hctx) const;
-  bool decrypt(const unsigned char* iv, EVP_CIPHER_CTX *ectx, HMAC_CTX *hctx) const;
+
+#if OPENSSL_VERSION_MAJOR >= 3
+  constexpr static const char* sha256KeyName{"sha256"};
+  constexpr static const size_t sha256KeyNameLength{6};
```

Comment:
Would you mind moving these to `libssl.cc` instead?

Reply:
why not just use std::string? you can then use c_str()

Reply:
> why not just use std::string? you can then use c_str()

and `.size()`.

Reply:
I don't think it's possible to use `std::string` with `constexpr`?
I suspect the compiler's constant folding mechanism should kick in, but I'm never sure in C++ so I figured it's just better to be explicit about what I want.

Reply:
`constexpr` strings are in C++20 I believe. But I prefer `const` over `constexpr` if it allows to enhance readability.